service/aggregator.go

```diff
@@ -113,23 +106,52 @@ func (s *AggregateSuggester) GetSuggestions(payload []byte, tid string) (Suggest
 	return aggregateResp, nil
 }
 
-func (s *AggregateSuggester) getConcordedSuggestions(tid string, suggestions []Suggestion) ([]Suggestion, error) {
-	logEntry := s.Log.WithTransactionID(tid)
+// suggesterErr is a thin wrapper error for marking the origin of an error
+// This error type is introduced as a hack in order to keep the service behavior, while changing the path of the data
```

Comment:
A suggestion for a clearer wording:

```suggestion
// This error type is introduced in order to allow us to distinguish between errors while keeping the original error
// ... because two errors cannot be wrapped with the standard library while keeping their original objects.
// https://golang.org/pkg/fmt/#Errorf says: "It is invalid to include more than one %w verb ... The %w verb is otherwise a synonym for %v."
```

Reply:
Both you and Eli had similar problems understanding the purpose of the new error type, so I've reworked the code a bit to address this issue.